src/App.js

```diff
@@ -1,13 +1,49 @@
 import React from 'react';
+import Todolist from './components/TodoComponents/TodoList';
+import TodoForm from './components/TodoComponents/TodoForm';
+
+const todo = [{
+  task: '',
+  id: Date.now(),
+  completed: false
+}];
 
 class App extends React.Component {
   // you will need a place to store your state in this component.
   // design `App` to be the parent component of your application.
   // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor() {
+    super();
+    this.state ={
+      todo: todo,
+    };
+  };
+
+  addTodo = event => {
+    console.log(event);
+    const newTodo = {
+      task: event,
+      id: Date.now(),
+      completed: false
+    }
+    // console.log(newTodo, ' newTodo')
+    this.setState({
+      todo: [...this.state.todo, newTodo]
+    });
+  };
+
+  handleComplete = () => {
+    this.setState({ completed : !this.state.completed })
```

Comment:
This is looking for state that looks like this: 
```
this.state = {
   todo: todo,
   completed: false
}
```
If that state existed, it would be state that just controls a broad "completed? yes/no" switch, and it wouldn't be linked to any _specific_ object or todo.
What you need is to make it look inside a single, specific object in the array and check that object's completed value. What's a good way to **identify** a single object in an array -- is there a property in each todo object you can use to point at that specific todo? Your function will need to take in that property so it knows which completed property to toggle.